sdk/typespec/typespec_client_core/src/error/http_error.rs

```diff
@@ -187,8 +187,12 @@ struct ErrorDetails {
 }
 
 impl ErrorDetails {
```

Comment:
Ultimately, this whole structure needs to be lifted up to `azure_core`. It's literally defined in the Azure REST API Guidelines in https://aka.ms/azapi/guidelines. Instead, what we could have is our Azure error info - much like this (still needs some tweaks) but then in `typespec::Error` maybe we just need a generic HTTP error that holds a `RawResponse` and the HTTP `StatusCode`.

Reply:
Discussed today as part of #2962 and not blocking this PR.